_extensions/acronyms/acronyms_styles.lua

```diff
@@ -139,33 +138,34 @@ return function(acronym, style_name, insert_links, is_first_use, plural,
     -- e.g., for plural form; and for sentence case
     acronym = acronym:clone()
     if plural then
+        -- Conditional strictness: when markdown is present in a part, require an explicit plural for that part.
+        local need_long_strict = Helpers.contains_markdown(acronym.longname) and not acronym._explicit_plural_longname
+        local need_short_strict = Helpers.contains_markdown(acronym.shortname) and not acronym._explicit_plural_shortname
+        if need_long_strict then
+            quarto.log.error("[acronyms] Plural form requested for '" .. tostring(acronym.key) .. "' but 'plural.longname' was not explicitly provided while markdown parsing is enabled for its longname. Define it under plural: { longname: ... } to use \\acrs{" .. tostring(acronym.key) .. "} .")
+            assert(false)
+        end
+        if need_short_strict then
+            quarto.log.error("[acronyms] Plural form requested for '" .. tostring(acronym.key) .. "' but 'plural.shortname' was not explicitly provided while markdown parsing is enabled for its shortname. Define it under plural: { shortname: ... } to use \\acrs{" .. tostring(acronym.key) .. "} .")
+            assert(false)
+        end
+        -- Apply plural forms (explicit provided parts already present; fallbacks safe for non-markdown components).
         acronym.shortname = acronym.plural.shortname
         acronym.longname = acronym.plural.longname
     end
 
-    if case == "upper" then
-        if case_target == "short" or case_target == "both" then
-            acronym.shortname = string.upper(acronym.shortname)
-        end
-        if case_target == "long" or case_target == "both" then
-            acronym.longname = string.upper(acronym.longname)
-        end
-    elseif case == "lower" then
-        if case_target == "short" or case_target == "both" then
-            acronym.shortname = string.lower(acronym.shortname)
-        end
-        if case_target == "long" or case_target == "both" then
-            acronym.longname = string.lower(acronym.longname)
-        end
-    elseif case == "sentence" then
-        if case_target == "short" or case_target == "both" then
-            acronym.shortname = capitalize_first(acronym.shortname)
-        end
-        if case_target == "long" or case_target == "both" then
-            acronym.longname = capitalize_first(acronym.longname)
-        end
+    -- Delegate case transformation to Helpers.transform_case which preserves inline formatting
+    local function transform_case(value, case_kind)
+        return Helpers.transform_case(value, case_kind)
+    end
+
+    if case_target == "short" or case_target == "both" then
+        acronym.shortname = transform_case(acronym.shortname, case)
+    end
+    if case_target == "long" or case_target == "both" then
+        acronym.longname = transform_case(acronym.longname, case)
     end
```

Comment:
Why use a local function that delegates to `Helpers`?

```suggestion
    if case_target == "short" or case_target == "both" then
        acronym.shortname = Helpers.transform_case(acronym.shortname, case)
    end
    if case_target == "long" or case_target == "both" then
        acronym.longname = Helpers.transform_case(acronym.longname, case)
    end
```

Reply:
Oops, this was a silly one!